x/tax2gas/post/post.go

```diff
@@ -75,11 +77,10 @@ func (tgd Tax2gasPostDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simulate
 	taxGas := ctx.TaxGasMeter().GasConsumed()
 
 	// we consume the gas here as we need to calculate the tax for consumed gas
-	ctx.GasMeter().ConsumeGas(taxGas, "tax gas")
```

Comment:
This will still be correct. I added them on gas remaining on the line below this

Reply:
Yes it is, as I said, tax gas is not related to gas limit, it's only related to fees. Therefore, the gas auto is still correct.

Reply:
Yes, we still check that the fees user send is enough to cover tax or not, but we will separate tax as taxgas, not gas. It's because as you said, if we combine them, gas for tx with amount larger than ~2.5b LUNC will be overflow. The `fees are not enough to pay for gas` is due to the money you send is not enogh to cover normal gas, which can work with `gas-multiplier`.

Reply:
You can't say tax2gas will not work just because you don't use `gas-multiplier`, that's completely offensive

Reply:
Hey you are right. I think in that case, we will use the normal gas meter to consume the tax gas again. With tx that have taxes larger than 2.4b lunc, which rarely happen, I think they will need to put the tx with exact fee amount, and they should do that by cli.

Reply:
I tried that before, but in [cosmos-sdk](https://github.com/classic-terra/cosmos-sdk/blob/d90304340869ede75ece46a91faf43cb25726a26/store/types/gas.go#L186) they will also panic if overflow